pkg/operator/provisioner/provisioner.go

```diff
@@ -141,38 +135,31 @@ func (p *RookVolumeProvisioner) Provision(options controller.VolumeOptions) (*v1
 }
 
 // createVolume creates a rook block volume.
-func createVolume(image, pool string, size int64, client rookclient.RookRestClient) (string, error) {
-	newImage := model.BlockImage{
-		Name:     image,
-		PoolName: pool,
-		Size:     uint64(size),
+func (p *RookVolumeProvisioner) createVolume(image, pool string, size int64) error {
+	if image == "" || pool == "" || size == 0 {
+		return fmt.Errorf("image missing required fields (image=%s, pool=%s, size=%d)", image, pool, size)
 	}
 
-	res, err := client.CreateBlockImage(newImage)
+	createdImage, err := ceph.CreateImage(p.context, p.provConfig.clusterName, image, pool, uint64(size))
 	if err != nil {
-		return "", fmt.Errorf("Failed to create rook block image %s/%s: %v", pool, image, err)
+		return fmt.Errorf("Failed to create rook block image %s/%s: %v", pool, image, err)
 	}
+	logger.Infof("Rook block image created: %s", createdImage.Name)
 
-	return res, nil
+	return nil
 }
 
 // Delete removes the storage asset that was created by Provision represented
 // by the given PV.
 func (p *RookVolumeProvisioner) Delete(volume *v1.PersistentVolume) error {
-	rookClient, err := api.GetRookClient(p.provConfig.clusterNamespace, p.client)
-	if err != nil {
-		return fmt.Errorf("Failed to get rook client: %v", err)
-	}
+	logger.Infof("Deleting volume %s", volume.Name)
 
-	image := model.BlockImage{
-		Name:     volume.Spec.PersistentVolumeSource.RBD.RBDImage,
-		PoolName: p.provConfig.pool,
-	}
-
-	_, err = rookClient.DeleteBlockImage(image)
+	name := volume.Spec.PersistentVolumeSource.RBD.RBDImage
+	err := ceph.DeleteImage(p.context, p.provConfig.clusterNamespace, name, p.provConfig.pool)
```

Comment:
right, i'll use clusterName for consistency. We really need to remove clusterNamespace since it represents the same thing as clusterName. clusterNamespace isn't being used